File: app/src/main/java/com/luismedinaweb/linker/PrefHelper.java
package com.luismedinaweb.linker;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by Luis on 4/11/2015.
 */
public class PrefHelper {

    public static final String defaultServer = "N/A";
    public static final String defaultIP = "N/A";
    public static final int defaultPort = -1;
    private static PrefHelper prefHelper;
    private SharedPreferences sharedPreferences;
    private int serverPort;
    private String serverName;
    private String ipAddress;

    private PrefHelper(Context context){
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        serverName = sharedPreferences.getString("server", defaultServer);
        serverPort = sharedPreferences.getInt("port", defaultPort);
        ipAddress = sharedPreferences.getString("ipAddress", defaultIP);
    }

    public static PrefHelper getInstance(Context context){
        if(prefHelper == null){
            prefHelper =  new PrefHelper(context);
        }
        return prefHelper;
    }

    public boolean haveValidServer(){
        if(serverName.equals(defaultServer) || serverPort == defaultPort){
            return false;
        }
        return true;
    }

    public int getServerPort() {
        return serverPort;
    }

    public void setServerPort(int serverPort) {
        sharedPreferences.edit().putInt("port", serverPort).commit();
        this.serverPort = serverPort;
    }

    public String getServerName() {
        return serverName;
    }

    public void setServerName(String serverName) {
        sharedPreferences.edit().putString("server", serverName.toUpperCase()).commit();
        this.serverName = serverName.toUpperCase();
    }

    public String getIpAddress() {
        return ipAddress;
    }

    public void setIpAddress(String ipAddress) {
        sharedPreferences.edit().putString("ipAddress", ipAddress).commit();
        this.ipAddress = ipAddress;
    }
}

File: app/src/main/java/com/luismedinaweb/linker/PrefHelper.kt
package com.luismedinaweb.linker

import android.content.SharedPreferences
import androidx.preference.PreferenceManager

object PrefHelper {

    private lateinit var sharedPreferences: SharedPreferences
    private const val IP_KEY: String = "ipAddress"
    private const val PORT_KEY: String = "port"

    var serverData: ServerData? = null
        set(value) {
            if (value != null) {
                sharedPreferences.edit().putInt(PORT_KEY, value.port).apply()
                sharedPreferences.edit().putString(IP_KEY, value.ipAddress).apply()
            } else {
                sharedPreferences.edit().clear().apply()
            }
            field = value
        }

    fun initialize(application: MainApplication) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(application)
        val port = sharedPreferences.getInt(PORT_KEY, 0)
        val ipAddress = sharedPreferences.getString(IP_KEY, null)
        if (port >= 0 && ipAddress != null) {
            serverData = ServerData(ipAddress, port)
        }
    }

    fun hasValidServer(): Boolean {
        return serverData != null
    }
}